packages/gatsby/src/schema/__tests__/build-schema.js

```diff
@@ -759,28 +761,32 @@ describe(`Build schema`, () => {
       const schema = await buildSchema()
       const nestedFields = schema.getType(`PluginDefinedNested`).getFields()
       const fields = schema.getType(`PluginDefined`).getFields()
-      expect(Object.keys(nestedFields)).toEqual([`foo`])
+      expect(Object.keys(nestedFields)).toEqual([`foo`, `bar`])
       expect(Object.keys(fields)).toEqual([
         `foo`,
         `baz`,
+        `bar`,
+        `qux`,
         `id`,
         `parent`,
         `children`,
         `internal`,
       ])
       expect(report.warn).toHaveBeenCalledWith(
-        `Plugin \`some-other-gatsby-plugin\` tried to define the GraphQL type ` +
+        `Plugin \`some-other-gatsby-plugin\` has customized the GraphQL type ` +
           `\`PluginDefinedNested\`, which has already been defined by the plugin ` +
           `\`some-gatsby-plugin\`.`
       )
       expect(report.warn).toHaveBeenCalledWith(
-        `Plugin \`some-other-gatsby-plugin\` tried to define the GraphQL type ` +
+        `Plugin \`some-other-gatsby-plugin\` has customized the GraphQL type ` +
           `\`PluginDefined\`, which has already been defined by the plugin ` +
           `\`some-gatsby-plugin\`.`
       )
     })
 
-    it(`does not merge plugin-defined type (graphql-js) with type defined by other plugin`, async () => {
+    // FIXME: Fails with: Error: Cannot get field 'foo' from type 'PluginDefinedNested'. Field does not exist.
+    //   same as "merges user-defined type (graphql-js) with plugin-defined type"
+    it.skip(`warns when merging plugin-defined type (graphql-js) with type defined by other plugin`, async () => {
```

Comment:
Turns out this use case is not supported. Same as here:

https://github.com/gatsbyjs/gatsby/blob/f1f312d74417ba9e1da845185df955650f76fb43/packages/gatsby/src/schema/__tests__/build-schema.js#L549-L550